nx/lib/nx/defn/grad.ex

```diff
@@ -446,6 +446,47 @@ defmodule Nx.Defn.Grad do
     [{x, reduce_g(x, opts, g)}]
   end
 
+  defp grad(:product, [x, opts], ans, g) do
+    axes =
+      case opts[:axes] do
+        [] -> Nx.axes(x.shape)
+        nil -> Nx.axes(x.shape)
+        axes -> axes
+      end
+
```

Comment:
```suggestion
    axes = opts[:axes] || Nx.axes(x)
```

@polvalente in the other place we do this:

https://github.com/elixir-nx/nx/blob/84407cf85244faa83ef22995bbee3c35d5ec2f2e/nx/lib/nx/defn/grad.ex#L574-L576

Interestingly, binary backend raises on empty list, EXLA returns the tensor unchanged and Torchx multiples over all axes. We should unify this edge case, I think it would make sense to validate the list is not empty in the Nx functions upfront. wdyt?

Reply:
Yes, this makes sense!